src/reducers/profile-view.js

```diff
@@ -169,6 +169,35 @@ function viewOptionsPerThread(state: ThreadViewOptions[] = [], action: Action) {
         ...state.slice(threadIndex + 1),
       ];
     }
+    case 'CHANGE_INVERT_CALLSTACK': {
+      const { callTree, callNodeTable, selectedThreadIndex } = action;
+      return state.map((viewOptions, threadIndex) => {
+        let selectedCallNodePath;
+        let expandedCallNodePaths;
+        if (selectedThreadIndex === threadIndex) {
+          // Only attempt this on the current thread, as we need the transformed thread
+          // There is no guarantee that this has been calculated on all the other threads,
+          // and we shouldn't attempt to expect it, as that could be quite a perf cost.
+          selectedCallNodePath = Transforms.invertCallNodePath(
+            viewOptions.selectedCallNodePath,
+            callTree,
+            callNodeTable
+          );
+
+          expandedCallNodePaths = [];
+          for (let i = 1; i < selectedCallNodePath.length; i++) {
+            expandedCallNodePaths.push(selectedCallNodePath.slice(0, i));
+          }
+        } else {
+          selectedCallNodePath = [];
+          expandedCallNodePaths = [];
+        }
+        return Object.assign({}, viewOptions, {
+          selectedCallNodePath,
+          expandedCallNodePaths,
+        });
```

Comment:
per the previous comment, this line should be moved inside the `if` block; in the `else`  block we should just return viewOptions.

Reply:
Done.